ambry-network/src/main/java/com.github.ambry.network/ResponseInfo.java

```diff
@@ -63,10 +63,16 @@ public NetworkClientErrorCode getError() {
   /**
    * @return the response received for this request.
    */
-  public ByteBuffer getResponse() {
+  public Object getResponse() {
```

Comment:
Is there an opportunity to use generics here? It's ok if it's not possible. I realize it may not be easily done since the buffer type is determined by a runtime config.

Reply:
Tried a couple ways to use generic, but not very successfully. They are either quite intrusive, or adding too many encapsulation.